clippy_dev/src/update_lints.rs

```diff
@@ -230,72 +227,75 @@ pub fn rename(old_name: &str, new_name: &str, uplift: bool) {
             .then_with(|| lhs.old_name.cmp(&rhs.old_name))
     });
 
+    write_file(
+        Path::new("clippy_lints/src/renamed_lints.rs"),
+        &gen_renamed_lints_list(&renamed_lints),
+    );
+
     if uplift {
+        write_file(Path::new("tests/ui/rename.rs"), &gen_renamed_lints_list(&renamed_lints));
         println!(
-            "`{}` has be uplifted. Please remove all the code inside `clippy_lints` related to it.",
+            "`{}` has be uplifted. All the code inside `clippy_lints` related to it needs to be removed manually.",
             old_name
         );
-        let contents = gen_renamed_lints_test(&renamed_lints);
-        fs::write("tests/ui/rename.rs", contents.as_bytes())
-            .unwrap_or_else(|e| panic!("error writing file `tests/ui/rename.rs`: {}", e));
     } else if found_new_name {
+        write_file(Path::new("tests/ui/rename.rs"), &gen_renamed_lints_list(&renamed_lints));
         println!(
-            "`{}` is already defined. The old linting code inside `clippy_lints` has to be updated/removed manually",
+            "`{}` is already defined. The old linting code inside `clippy_lints` needs to be updated/removed manually.",
             new_name
         );
-        let contents = gen_renamed_lints_test(&renamed_lints);
-        fs::write("tests/ui/rename.rs", contents.as_bytes())
-            .unwrap_or_else(|e| panic!("error writing file `tests/ui/rename.rs`: {}", e));
     } else {
         // Rename the lint struct and the containing module if it shares the lint name across all files.
         let lint = &mut lints[old_lint_index];
         let old_name_upper = old_name.to_uppercase();
         let new_name_upper = new_name.to_uppercase();
-        let new_module_path = PathBuf::from(format!("clippy_lints/src/{}.rs", new_name));
-        let replacements;
 
-        let (replacements, mut renamed_file) = if lint.module == lint.name
-            // Only rename the module if the file doesn't already exist.
-            && let Ok(file) = fs::OpenOptions::new().create_new(true).write(true).open(&new_module_path)
-        {
+        if try_rename_file(
+            Path::new(&format!("tests/ui/{}.rs", old_name)),
+            Path::new(&format!("tests/ui/{}.rs", new_name)),
```

Comment:
Is there an advantage in checking each path individually instead of iterating over every file and then only checking the file name regardless of the path?

Reply:
This is the easiest way to make sure the new name doesn't already exist before we start making changes based on the rename being successful. e.g. `tests/ui/foo.stderr` shouldn't be renamed to `tests/ui/bar.stderr` if `tests/ui/bar.rs` already exists.

A similar issue happens with module names